pkg/apis/servicecatalog/validation/instance.go

```diff
@@ -254,11 +255,26 @@ func validateServiceInstanceUpdate(instance *sc.ServiceInstance) field.ErrorList
 }
 
 // internalValidateServiceInstanceUpdateAllowed ensures there is not a
-// pending update on-going with the spec of the instance before allowing an update
-// to the spec to go through.
-func internalValidateServiceInstanceUpdateAllowed(new *sc.ServiceInstance, old *sc.ServiceInstance) field.ErrorList {
+// pending update on-going with the spec of the instance before allowing an
+// update to the spec to go through unless its the same user who made the
+// original update.
+func internalValidateServiceInstanceUpdateAllowed(ctx genericapirequest.Context, new *sc.ServiceInstance, old *sc.ServiceInstance) field.ErrorList {
 	errors := field.ErrorList{}
-	if old.Generation != new.Generation && old.Status.CurrentOperation != "" {
+
+	oldUID := ""
+	newUID := ""
+
+	if old.Spec.UserInfo != nil {
+		oldUID = old.Spec.UserInfo.UID
+	}
+
+	if ctx != nil {
+		if user, _ := genericapirequest.UserFrom(ctx); user != nil {
```

Comment:
do we have any idea if this works at all with our apiserver? Does this code only work when originating-identity is turned on? Is that the only case we're concerned with. I think so.

Reply:
I assume the lock is on all the time.  This does seem to work with our apiserver since the test appears to pass :-)